src/main/java/org/team1540/robot2024/subsystems/vision/AprilTagVision.java

```diff
@@ -0,0 +1,84 @@
+package org.team1540.robot2024.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+import org.team1540.robot2024.Constants;
+import org.team1540.robot2024.util.vision.TimestampedVisionPose;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+public class AprilTagVision extends SubsystemBase {
+    private final AprilTagVisionIO frontCameraIO;
+    private final AprilTagVisionIOInputsAutoLogged frontCameraInputs = new AprilTagVisionIOInputsAutoLogged();
+
+    private final AprilTagVisionIO rearCameraIO;
+    private final AprilTagVisionIOInputsAutoLogged rearCameraInputs = new AprilTagVisionIOInputsAutoLogged();
+
+    private final Consumer<TimestampedVisionPose> visionPoseConsumer;
+
+    private TimestampedVisionPose frontPose =
+            new TimestampedVisionPose(-1, new Pose2d(), new int[0], new Pose2d[0]);
+    private TimestampedVisionPose rearPose =
+            new TimestampedVisionPose(-1, new Pose2d(), new int[0], new Pose2d[0]);
+
+    public AprilTagVision(
+            AprilTagVisionIO frontCameraIO,
+            AprilTagVisionIO rearCameraIO,
+            Consumer<TimestampedVisionPose> visionPoseConsumer) {
+        this.frontCameraIO = frontCameraIO;
+        this.rearCameraIO = rearCameraIO;
+        this.visionPoseConsumer = visionPoseConsumer;
+    }
+
+    @Override
+    public void periodic() {
+        frontCameraIO.updateInputs(frontCameraInputs);
+        rearCameraIO.updateInputs(rearCameraInputs);
+        Logger.processInputs("Vision/FrontCamera", frontCameraInputs);
+        Logger.processInputs("Vision/RearCamera", rearCameraInputs);
+
+        if (frontCameraInputs.lastMeasurementTimestampSecs > frontPose.timestampSecs()) {
```

Comment:
When would this not be the case

Reply:
When the IO has detected a more recent pose and the latest pose field has to be updated

Reply:
if io has detected a more recent pose, this runs, but when would io not have detected a more recent pose?

Reply:
If the limelight frame rate is less than 50 fps the IO will not have detected a more recent pose for some loop cycles.